plugin/mlsdbonlinelocator.cpp

```diff
@@ -311,6 +335,26 @@ bool MlsdbOnlineLocator::readServerResponseData(const QByteArray &data, QString
     return true;
 }
 
+void MlsdbOnlineLocator::checkError(const QByteArray &data)
+{
+    QJsonParseError parseError;
+    QJsonDocument json = QJsonDocument::fromJson(data, &parseError);
+    if (parseError.error != QJsonParseError::NoError) {
+        return;
+    }
+
+    if (!json.isObject()) {
+        return;
+    }
+
+    int errorCode = json.object().value(QLatin1String("error")).toObject().value(QLatin1String("code")).toInt();
+
+    if (errorCode == 400) {
+        qCDebug(lcGeoclueMlsdbOnline) << "Detected invalid API key, disabling locator for some time";
```

Comment:
I would print a warning instead and even if you haven't enabled mls provider debugs. Also be less ambigious i.e. "Mozilla Location Service online positioning failed due to missing or invalid key, disabling locator for 12 hours"